File: test/rust/crates/specgate-fixtures/src/conformance/async/async_fetch.rs
// Async operation — tests that harness generates async runner. The body awaits
// an immediately-ready future so the `await` machinery is genuinely exercised
// without needing a reactor; the C# conformance mirror awaits `Task.FromResult`.
use specgate::*;

#[spec_operation("fetch", spec = "fixture.async_fetch")]
pub async fn fetch(url: &str) -> String {
    std::future::ready(format!("response from {url}")).await
}

File: test/csharp/SpecGate.CtscFixtures/Conformance/Async/AsyncFetch.cs
using SpecGate.Annotations;

namespace SpecGate.CtscFixtures.Conformance.Async;

public static class AsyncFetch
{
    [SpecOperation("fetch", Spec = "fixture.async_fetch")]
    public static async Task<string> Fetch([SpecInput("url")] string url) =>
        await Task.FromResult($"response from {url}");
}
